openc3/lib/openc3/models/python_package_model.rb

```diff
@@ -71,16 +74,23 @@ def self.install(name_or_path, scope:)
       package_filename = File.basename(package_file_path)
       begin
         pypi_url = get_setting('pypi_url', scope: scope)
-      rescue
-        # If Redis isn't running try the ENV, then simply pypi.org/simple
-        pypi_url = ENV['PYPI_URL']
         if pypi_url
           pypi_url += '/simple'
         end
-        pypi_url ||= 'https://pypi.org/simple'
+      rescue => e
+        Logger.error("Failed to retrieve pypi_url: #{e.formatted}")
+      ensure
+        if pypi_url.nil?
+          # If Redis isn't running try the ENV, then simply pypi.org/simple
+          pypi_url = ENV['PYPI_URL']
+          if pypi_url
+            pypi_url += '/simple'
+          end
+          pypi_url ||= 'https://pypi.org/simple'
+        end
```

Comment:
Logic is cleaner but basically the same?

Reply:
Basically the same.